precompiles/vesting/integration_test.go

```diff
@@ -234,411 +286,479 @@ var _ = Describe("Interacting with the vesting extension", func() {
-				Expect(balancePre.Amount).To(Equal(math.NewInt(1100)), "expected different balance after setup")
-
-				clawbackArgs := s.BuildCallArgs(callType, contractAddr).
-					WithMethodName(vesting.ClawbackMethod).
-					WithArgs(
-						s.address,
-						toAddr,
-						differentAddr,
-					)
+				res, err := s.grpcHandler.GetBalance(s.keyring.GetAccAddr(vestingAccIdx), s.bondDenom)
+				Expect(err).To(BeNil())
+				balancePre := res.Balance
+
+				callArgs, txArgs := s.BuildCallArgs(callType, contractAddr)
+				callArgs.MethodName = vesting.ClawbackMethod
+				callArgs.Args = []interface{}{
+					s.keyring.GetAddr(0),
+					s.keyring.GetAddr(vestingAccIdx),
+					differentAddr,
+				}
 
 				clawbackCheck := passCheck.
 					WithExpEvents(vesting.EventTypeClawback)
 
-				_, ethRes, err := contracts.CallContractAndCheckLogs(s.ctx, s.app, clawbackArgs, clawbackCheck)
+				_, ethRes, err := s.factory.CallContractAndCheckLogs(s.keyring.GetPrivKey(0), txArgs, callArgs, clawbackCheck)
 				Expect(err).ToNot(HaveOccurred(), "error while calling the contract: %v", err)
+				Expect(s.network.NextBlock()).To(BeNil())
+
+				var (
+					co             vesting.ClawbackOutput
+					expClawbackAmt = math.NewInt(1000)
+				)
 
-				var co vesting.ClawbackOutput
 				err = s.precompile.UnpackIntoInterface(&co, vesting.ClawbackMethod, ethRes.Ret)
 				Expect(err).ToNot(HaveOccurred(), "error while unpacking the clawback output: %v", err)
 				Expect(co.Coins).To(Equal(balances), "expected different clawback amount")
 
-				balancePost := s.app.BankKeeper.GetBalance(s.ctx, toAddr.Bytes(), s.bondDenom)
-				Expect(balancePost.Amount.Int64()).To(Equal(int64(100)), "expected only initial balance after clawback")
-				balanceReceiver := s.app.BankKeeper.GetBalance(s.ctx, differentAddr.Bytes(), s.bondDenom)
-				Expect(balanceReceiver.Amount).To(Equal(math.NewInt(1000)), "expected receiver to show different balance after clawback")
+				res, err = s.grpcHandler.GetBalance(s.keyring.GetAccAddr(vestingAccIdx), s.bondDenom)
+				Expect(err).To(BeNil())
+				balancePost := res.Balance
+				Expect(balancePost.Amount).To(Equal(balancePre.Amount.Sub(expClawbackAmt)), "expected only initial balance after clawback")
+				res, err = s.grpcHandler.GetBalance(differentAddr.Bytes(), s.bondDenom)
+				Expect(err).To(BeNil())
+				balanceReceiver := res.Balance
+				Expect(balanceReceiver.Amount).To(Equal(expClawbackAmt), "expected receiver to show different balance after clawback")
 			})
 
 			It(fmt.Sprintf("should return an error when not sending as the funder (%s)", callType.name), func() {
 				// create and fund new account
 				differentAddr, differentPriv := testutiltx.NewAddrKey()
-				err := evmosutil.FundAccountWithBaseDenom(s.ctx, s.app.BankKeeper, differentAddr.Bytes(), 1e18)
+				err := testutils.FundAccountWithBaseDenom(
+					s.factory, s.network,
+					s.keyring.GetKey(0),
+					differentAddr.Bytes(),
+					math.NewInt(1e18),
+				)
```

Comment:
also here, relying on funding the address during tests can be avoided as it's not a "real-world" operation during the integration tests

```suggestion
				differentSender := s.keyring.GetKey(2)
```